File: travel-management/backend/domain/model/hotel.go
package model

import "encoding/json"

type Hotel struct {
	Id          uint       `json:"id"`
	Name        string     `json:"name"`
	Address     Address    `json:"address"`
	Description string     `json:"description"`
	Vendor      Vendor     `json:"vendor"`
	Pictures    []*Picture `json:"pictures"`
	Travels     []*Travel  `json:"Travels"`
}

func (h *Hotel) String() string {
	res, err := json.Marshal(h)
	if err != nil {
		return ""
	}
	return string(res)
}

File: travel-management/frontend/src/app/models/hotel.ts
import { Tag } from './tag';
import { Picture } from './pictures';
import { Travel } from './travel';

export interface Hotel {
  id: number;
  hotelname: string;
  street: string;
  state: string;
  land: string;
  vendorid: number;
  vendorname: string;
  description: string;
  pictures: Picture[];
  tags: Tag[];
  travels: Travel[];
}
